vmm/src/default_syscalls.rs

```diff
@@ -95,11 +99,31 @@ const MAP_PRIVATE: u64 = 0x02;
 const MAP_ANONYMOUS: u64 = 0x20;
 const MAP_NORESERVE: u64 = 0x4000;
 
+/// Applies the configured level of seccomp filtering to the current thread.
+pub fn set_seccomp_level(seccomp_level: u32) -> Result<(), Error> {
+    // Load seccomp filters before executing guest code.
+    // Execution panics if filters cannot be loaded, use --seccomp-level=0 if skipping filters
+    // altogether is the desired behaviour.
+    match seccomp_level {
+        SECCOMP_LEVEL_ADVANCED => setup_seccomp(SeccompLevel::Advanced(default_context()?)),
+        SECCOMP_LEVEL_BASIC => setup_seccomp(seccomp::SeccompLevel::Basic(ALLOWED_SYSCALLS)),
+        SECCOMP_LEVEL_NONE | _ => Ok(()),
+    }
+}
+
 /// The default context containing the white listed syscall rules required by `Firecracker` to
 /// function.
 pub fn default_context() -> Result<SeccompFilterContext, Error> {
     Ok(SeccompFilterContext::new(
         vec![
+            (
+                libc::SYS_accept,
```

Comment:
Would it make sense to choke the vcpus as much as possible and have 2 sets of default filters? Who needs `accept`, `tkill` and `clock_gettime`? If it's only the vmm thread, the vcpus can be restricted even more. Customer code that ends up calling `tkill` might be dangerous.

Reply:
I was thinking about three sets actually VMM, API and VCPU. But we can iterate on that after.
There is also the question of whether that is actually useful, an attacker gaining control of one thread can then easily influence the execution of another thread in the same process, therefore per-thread seccomp filtering strategies wouldn't bring much value.